### DISCUSSION TITLE
ST7920 (LCD12864) 128x64 Support?

### DISCUSSION THREAD

**Discussion by Knucklez81** [davetcc/tcMenu#373] [Q&A] [answered]
Been trying TCMenu but having problems with it displaying - all I managed to get displayed is half the menu title.  In TCMenu I cannot find U8G2_ST7920_128X64_1_SW_SPI so have been trying all other options that are close and entering the SPI pins (and sometimes changing the constructor in the code) but so far nothing has worked yet works using just U8G2 in other projects.  Is the ST7920 (LCD12854) powered displays not supported? as its quite a common display.

Thanks.

**Comment by davetcc** [comment]
I've not actually come across one of those displays, but if you can get it working with U8G2 in a standalone sketch, then it should work with TcMenu without issue.

The U8G2 quick start plugin doesn't support all displays, but there is also a "Custom U8G2" option, and this supports all possible options, as you provide the constructed object name that you have already configured and set up yourself.

**Comment by Knucklez81** [comment]
After a bit of testing I have successfully got TCMenu to work on the ST7920 LCD's (Common on 3D Printers).

First I chose the U8G2 Display Quickstart and left everything (pins etc) blank as changed these in the code.  Then I opened the projects .CPP and .H file.  In the .cpp files I changed `U8G2_SSD1306_128X64_NONAME_F_4W_HW_SPI gfx(U8G2_R0, U8X8_PIN_NONE, U8X8_PIN_NONE, U8X8_PIN_NONE);` 

to:

`U8G2_ST7920_128X64_F_SW_SPI gfx(U8G2_R0, 23, 17,16); (pins relevant for my setup, enabled Full Buffer otherwise if not full buffer only one line will show)`

and in the .h file changed `U8G2_SSD1306_128X64_NONAME_F_4W_HW_SPI` to `U8G2_ST7920_128X64_F_SW_SPI`.

Also - had to check "turn on support for UTF-8, and Un-check "Use TcUnicode".

Saved, Compiled and uploaded and it displays and functions perfectly.

So in saying that, maybe add in an option for this display using U8G2_ST7920_128X64_F_SW_SPI in the dropdown?

**Reply by davetcc** [reply]
See https://github.com/davetcc/tcMenu/issues/375

**Reply by davetcc** [reply, marked as answer]
should be available in 4.1
